### PR TITLE
use opername instead of nickname when logging mask changes

### PR DESCRIPTION
**Pull request by examknow**
this is achieved by sending the opername instead of the nickname for the who parameter. per #7.

### CODE REVIEW

**Comment by jesopo** [comment]
see https://github.com/Libera-Chat/maskwatch-bot/issues/7#issuecomment-855274196

**Comment by examknow** [comment]
forgot to mention, to update the db tables:

    ALTER TABLE changes ADD COLUMN by_oper TEXT;
    ALTER TABLE changes RENAME COLUMN by TO by_nick;

**Comment by examknow** [comment]
Should be all ready now. In a future PR, I'd like to make the oper whois regexes configurable so that we can continue to easily support other ircds.

File: maskwatch-bot/__init__.py [hunk 1]
```diff
@@ -88,18 +90,21 @@ async def _oper_up(self,
...
+        #:lithium.libera.chat 320 sandcat sandcat :is opered as sandcat, privset sandcat
         #:lithium.libera.chat 318 sandcat sandcat :End of /WHOIS list.
 
         whois_line = await self.wait_for({
             whois_end, whois_oper
         })
-        return whois_line.command == RPL_WHOISOPERATOR
+        # return the oper name or nothing
+        if whois_line.command == RPL_WHOISSPECIAL and whois_line.params[2].startswith("is opered as"):
+            return whois_line.params[2].split(",")[0].split()[-1]
```
Comment (jesopo) [inline maskwatch-bot/__init__.py:105]: you won't always be told the privset, so sometimes you'll get
```
is opered as sandcat, privset sandcat
```
and sometimes you'll get
```
is opered as sandcat
```

so I'd instead suggest this
```python3
if whois_line.command == RPL_WHOISOPERATOR:
    match = re.search(r"^is opered as (\S+)(?:,|$)", whois_line.params[2])
    if match is not None:
        return match.group(1)
```

File: maskwatch-bot/__init__.py [hunk 2]
```diff
@@ -88,18 +90,21 @@ async def _oper_up(self,
...
-        #:lithium.libera.chat 313 sandcat sandcat :is an IRC Operator
+        #:lithium.libera.chat 320 sandcat sandcat :is opered as sandcat, privset sandcat
         #:lithium.libera.chat 318 sandcat sandcat :End of /WHOIS list.
 
         whois_line = await self.wait_for({
             whois_end, whois_oper
         })
-        return whois_line.command == RPL_WHOISOPERATOR
+        # return the oper name or nothing
+        if whois_line.command == RPL_WHOISSPECIAL and whois_line.params[2].startswith("is opered as"):
```
Comment (jesopo) [inline maskwatch-bot/__init__.py:104]: I think this still wants to be `RPL_WHOISOPERATOR`

**Review by jesopo — CHANGES_REQUESTED** [review]

File: maskwatch-bot/__init__.py [hunk 3]
```diff
@@ -88,18 +89,24 @@ async def _oper_up(self,
...
         #:lithium.libera.chat 318 sandcat sandcat :End of /WHOIS list.
 
         whois_line = await self.wait_for({
             whois_end, whois_oper
         })
-        return whois_line.command == RPL_WHOISOPERATOR
+        # return the oper name or nothing
+        if whois_line.command == RPL_WHOISOPERATOR:
+            match = re.search(r"^is opered as (\S+)(?:,|$)", whois_line.params[2])
+            print(match)
```
Comment (jesopo) [inline maskwatch-bot/__init__.py:106]: the curse of the forgotten debug print

File: maskwatch-bot/__init__.py [hunk 4]
```diff
@@ -329,25 +336,26 @@ async def line_read(self, line: Line):
...
             args:    str):
 
-        if await self._is_oper(who):
+        opername = await self._get_oper(who.nickname)
+        if opername is not None:
+            opername = None if opername == "<grant>" else opername
             attrib  = f"cmd_{command}"
             if hasattr(self, attrib):
-                outs = await getattr(self, attrib)(who, args)
+                outs = await getattr(self, attrib)(opername, who._source, args)
```
Comment (jesopo) [inline maskwatch-bot/__init__.py:354]: `str(who)` instead of `who._source`

**Pull request merged**